daemon/daemon.go

```diff
@@ -445,7 +445,23 @@ func (daemon *Daemon) DaemonLeavesCluster() {
 	// Daemon is in charge of removing the attachable networks with
 	// connected containers when the node leaves the swarm
 	daemon.clearAttachableNetworks()
+	// We no longer need the cluster provider, stop it now so that
+	// the network agent will stop listening to cluster events.
```

Comment:
I noticed that `setClusterProvider` calls `daemon.netController.SetClusterProvider`, which writes to the `c.cfg.Daemon.DisableProvider` channel. Since this is asynchronous, do we have to wait for the message sent on that channel to be received? It seems like it's important that the cluster provider be disabled before moving on to `ReleaseIngress`, because when these were in the opposite order the panic still happened.